templatestore/views.py

```diff
@@ -617,13 +646,21 @@ def patch_attributes_view(request, name):
             if "attributes" not in data:
                 raise (Exception("Validation: Attributes are not provided"))
 
-            template = Template.objects.filter(name=name)
-            if not len(template):
+            try:
+                template = Template.objects.get(name=name)
+            except Exception as e:
                 raise (Exception("Validation: Template with given name does not exist"))
 
+            cfgs = TemplateConfig.objects.filter(type=template.type)
+
             missing_attributes = set(
-                ts_settings.TE_TEMPLATE_ATTRIBUTES_KEYS
+                ts_settings.TE_TEMPLATE_ATTRIBUTES.keys()
+            ).difference(set(data["attributes"].keys()))
+
+            missing_attributes = missing_attributes | set(
+                cfgs[0].attributes.keys()
             ).difference(set(data["attributes"].keys()))
```

Comment:
what?

Reply:
this checks that no mandatory attribute is missing in the attributes given by the user.